pyxdf/pyxdf.py

```diff
@@ -368,15 +366,28 @@ def load_xdf(filename,
     return streams, fileheader
 
 
-def open_xdf(filename):
-    """Open XDF file for reading."""
-    filename = Path(filename)  # ensure convert to pathlib object
-    if filename.suffix == '.xdfz' or filename.suffixes == ['.xdf', '.gz']:
-        f = gzip.open(str(filename), 'rb')
+def open_xdf(file):
+    """Open XDF file for reading.
+    :type file: str | pathlib.Path | io.RawIOBase
+        File name or already opened file
+    """
+
+    if isinstance(file, (io.RawIOBase, io.BufferedIOBase)):
+        if isinstance(file, io.TextIOBase):
+            raise ValueError('file has to be opened in binary mode')
+        f = file
     else:
-        f = open(str(filename), 'rb')
+        filename = Path(file).resolve()  # absolute path following symlinks
+        if not filename.exists():
+            raise Exception('file %s does not exist.' % filename)
+
+        filename = Path(file)  # ensure convert to pathlib object
```

Comment:
I'd do `filename = Path(file)` in line 380, and then check `if not filename.resolve().exists():` in order to avoid assigning to `filename` twice.

Reply:
Done.